source/common/http/conn_manager_impl.cc

```diff
@@ -1728,6 +1728,15 @@ bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
   if (state_.created_filter_chain_) {
     return false;
   }
+
+  // Ignore h2c upgrade requests until we support them.
+  // See https://github.com/envoyproxy/envoy/issues/7161 for details.
+  if (request_headers_ && request_headers_->Upgrade() &&
```

Comment:
so we can do this here, but as I think it only makes sense to do this for HTTP/1 I was thinking we could clean them before the HCM even sees them, say in onHeadersComplete in source/common/http/http1/codec_impl.cc

Arguably we shouldn't get an h2c upgrade over H2 or QUIC and if we do, I'm happy to send a 500 "what are you doing".  What do you think?

Reply:
Makes sense.  I was thinking that this was the place where the upgrade are handled now I see that source/common/http/http1/codec_impl.cc has some logic as well.  Will do.

Reply:
Should I remove the Connection: upgrade header as well or leave it, and if so what is the best way to remove a single value from a header.  Connection can have two values, e.g. Connection: keep-alive, upgrade see test/integration/websocket_integration_test.cc

Reply:
Done as discussed.  PTAL